src/Main.java

```diff
@@ -1,5 +1,314 @@
+               } else if (temporaryChange / twoDollarBill > amountOfTwos){
+                   temporaryChange = temporaryChange - amountOfTwos * twoDollarBill;
+                   amountOfTwos = 0;
+               }
+           }
+           if(amountOfOnes > 0) {
+               if (temporaryChange / oneDollarBill <= amountOfOnes) {
+                   amountOfOnes = amountOfOnes - temporaryChange / oneDollarBill;
+                   temporaryChange = temporaryChange - (temporaryChange / oneDollarBill) * oneDollarBill;
+               } else if (temporaryChange / oneDollarBill > amountOfOnes){
+                   temporaryChange = temporaryChange - amountOfOnes * oneDollarBill;
+                   amountOfOnes = 0;
+               }
+           }
+           if(temporaryChange == 0) {
+               totalAmountOfMoney = amountOfTwenties * twentyDollarBill + amountOfTens * tenDollarBill + amountOfTwos * twoDollarBill + amountOfOnes * oneDollarBill;
+               System.out.println("Money in cash register after returning change: " + totalAmountOfMoney + "\nBanknotes: " + "\n20 KM - " + amountOfTwenties + "\n10 KM - " + amountOfTens + "\n2 KM - " + amountOfTwos + "\n1 KM - " + amountOfOnes);
+           } else {
+               System.out.println("Sorry the cash register does not have the required amount of money.");
+           }
+       }
+ */
+
+       // Task1 + Bonus + Super Duper Sega Mega Bonus
+       long oneDollarBill = 1;
+       long twoDollarBill = 2;
+       long tenDollarBill = 10;
+       long twentyDollarBill = 20;
+       double fiftyCents = 0.5;
+       double twentyCents = 0.2;
+       double tenCents = 0.1;
+       double oneCent = 0.01;
+       long amountOfTwenties = 1;
+       long amountOfTens = 5;
+       long amountOfTwos = 5;
+       long amountOfOnes = 2;
+       long amountOfFiftyCents = 5;
+       long amountOfTwentyCents = 3;
+       long amountOfTenCents = 3;
+       long amountOfOneCent = 9;
+       double totalAmountOfMoney = amountOfTwenties * twentyDollarBill + amountOfTens * tenDollarBill + amountOfTwos * twoDollarBill + amountOfOnes * oneDollarBill + amountOfFiftyCents * fiftyCents + amountOfTwentyCents * twentyCents + amountOfTenCents * tenCents + amountOfOneCent * oneCent;
+       double spentMoney = 35.78;
+       double maximumAmountOfMoney = 100;
+       double totalAmountOfChange = Math.round((totalAmountOfMoney - spentMoney) * 100.0) / 100.0;
+        System.out.println("Task1 + Bonus + Super Duper Sega Mega Bonus: ");
+       System.out.println("Money in cash register before returning change: " + totalAmountOfMoney + "\nBanknotes in register: " + "\n20 KM - " + amountOfTwenties + "\n10 KM - " + amountOfTens + "\n2 KM - " + amountOfTwos + "\n1 KM - " + amountOfOnes + "\n0.50 KM - " + amountOfFiftyCents + "\n0.20 KM - " + amountOfTwentyCents + "\n0.10 KM - " + amountOfTenCents + "\n0.01 KM - " + amountOfOneCent + "\nTotal bill: " + spentMoney + " KM\nChange: " + totalAmountOfChange + " KM\n");
```

Comment:
Long output you could have formated the output better by using multiple printlnsm or even prepared whole logical sequences, the text can also be in external variables